pat/ReferenceTestV2/consumer/result_collector.py

```diff
@@ -0,0 +1,25 @@
+"""Result collector for test results."""
+
+class _ResultCollector:
+    def __init__(self):
+        self._results: list[str] = []
+        self._failed: bool = False
+
+    def log_success(self, step: str, message: str) -> None:
+        self._results.append(f'Success: {step} - {message}')
+
+    def log_failure(self, step: str, message: str) -> None:
+        self._results.append(f'Failure: {step} - {message}')
+        self._failed = True
+
+    @property
+    def failed(self) -> bool:
+        return self._failed
```

Comment:
would suggest to do:

`return self._failed or not self._results`

to be sure that at least one result was generated